packages/vite/src/node/plugins/resolve.ts

```diff
@@ -528,21 +549,33 @@ export function tryNodeResolve(
     basedir = nestedResolveFrom(nestedRoot, basedir, options.preserveSymlinks)
   }
 
-  const pkg = resolvePackageData(pkgId, basedir, options.preserveSymlinks)
+  let pkg: PackageData | undefined
+  const pkgId = possiblePkgIds.reverse().find((pkgId) => {
+    pkg = resolvePackageData(pkgId, basedir, options.preserveSymlinks)
+    return pkg
```

Comment:
Shouldnt it be `pkgId` here?

Reply:
Nope, we're using `find` to look for the first existing package.json

We could do `pkg !== undefined` to be more clear though

Reply:
Oh, ya. Sorry for the noise.